pandas/core/arrays/arrow/array.py

```diff
@@ -48,16 +53,149 @@
 
 class ArrowExtensionArray(ExtensionArray):
     """
-    Base class for ExtensionArray backed by Arrow array.
+    Base class for ExtensionArray backed by Arrow ChunkedArray.
     """
 
     _data: pa.ChunkedArray
 
-    def __init__(self, values: pa.ChunkedArray) -> None:
-        self._data = values
+    def __init__(self, values: pa.Array | pa.ChunkedArray) -> None:
+        if pa_version_under1p01:
+            msg = "pyarrow>=1.0.0 is required for PyArrow backed ArrowExtensionArray."
+            raise ImportError(msg)
+        if isinstance(values, pa.Array):
+            self._data = pa.chunked_array([values])
+        elif isinstance(values, pa.ChunkedArray):
+            self._data = values
+        else:
+            raise ValueError(
+                f"Unsupported type '{type(values)}' for ArrowExtensionArray"
+            )
+        self._dtype = ArrowDtype(self._data.type)
+
+    @classmethod
+    def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy=False):
+        """
+        Construct a new ExtensionArray from a sequence of scalars.
+        """
+        if isinstance(dtype, ArrowDtype):
+            pa_dtype = dtype.pa_dtype
+        elif dtype:
+            pa_dtype = pa.from_numpy_dtype(dtype)
+        else:
+            pa_dtype = None
+
+        if isinstance(scalars, cls):
+            data = scalars._data
+            if pa_dtype:
+                data = data.cast(pa_dtype)
+            return cls(data)
+        else:
+            # https://issues.apache.org/jira/browse/ARROW-16645
+            mask = [scalar is pa.NA or isna(scalar) for scalar in scalars]
+            return cls(pa.chunked_array(pa.array(scalars, mask=mask, type=pa_dtype)))
```

Comment:
Ah, I see you handled this with a mask. But so as mentioned in the previous comment, if you pass `from_pandas=True` to `pa.array`, pyarrow should handle any possible pandas missing value (np.nan, None, pd.NA, pd.NaT)